src/lib/Microsoft.Fx.Portability/Proxy/ProxyAuthenticationHandler.cs

```diff
@@ -41,7 +43,10 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             {
                 try
                 {
-                    return await base.SendAsync(request, cancellationToken);
+                    HttpResponseMessage result = await base.SendAsync(request, cancellationToken);
+                    this.LastStatusCode = result.StatusCode;
+                    result.EnsureSuccessStatusCode();
```

Comment:
It looks like we don't normally throw here for failing status code, so I'd like to avoid throwing for some condition beyond the proxy situation. This code path has likely regressed over time, and was clearly designed to handle exceptions, but I'd like to not perturb the way it works outside of the proxy failure.